src/boltz/data/parse/schema.py

```diff
@@ -785,7 +790,31 @@ def parse_boltz_schema(  # noqa: C901, PLR0915, PLR0912
     pocket_residues = []
     constraints = schema.get("constraints", [])
     for constraint in constraints:
-        if "bond" in constraint:
+        if "cyclic" in constraint:
+            sequence_items = schema.get("sequences", [])
+            if constraint["cyclic"] in list(chains.keys()):
+                for sequent_item in sequence_items:
+                    if "protein" not in sequent_item:
+                        msg = f"Cyclic constraints are supported only on proteins. Got {sequent_item}"
+                        raise ValueError(msg)
```

Comment:
This is more like what I was thinking. Note not tested at all and needs flushing out.

```suggestion
        if "cyclic" in constraint:
            sequence_items = schema.get("sequences", {}) # pretty sure this should be a dict but need to check yaml parsing
            cyclic_chain_id = constraint["cyclic"]
            if cyclic_chain_id not in list(chains.keys()):
                raise ValueError() # some useful error about chain_id not found
            
            for sequence_type, sequence in sequence_items.items():
                if cyclic_chain_id == sequence["id"] and sequence_type != "protein":
                    msg = f"Cyclic constraints are supported only on proteins. Got {sequence}"
                    raise ValueError(msg)
```